cloud/amazon/sts_assume_role.py

```diff
@@ -0,0 +1,166 @@
+    aliases: []
+  policy:
+    description:
+      - Supplemental policy to use in addition to assumed role's policies. 
+    required: false
+    default: null
+    aliases: []
+  duration_seconds:
+    description:
+      - The duration, in seconds, of the role session. The value can range from 900 seconds (15 minutes) to 3600 seconds (1 hour). By default, the value is set to 3600 seconds. 
+    required: false
+    default: null
+    aliases: []
+  external_id:
+    description:
+      - A unique identifier that is used by third parties to assume a role in their customers' accounts. 
+    required: false
+    default: null
+    aliases: []
+  mfa_serial_number:
+    description:
+      - he identification number of the MFA device that is associated with the user who is making the AssumeRole call. 
+    required: false
+    default: null
+    aliases: []
+  mfa_token:
+    description:
+      - The value provided by the MFA device, if the trust policy of the role being assumed requires MFA. 
+    required: false
+    default: null
+    aliases: []
+
+author: Boris Ekelchik
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Basic example of assuming a role
+tasks:
+- name: assume a role in account 123456789012
+  sts_assume_role: role_arn="arn:aws:iam::123456789012:role/someRole" session_name="someRoleSession"
+
+- name: display temporary credentials
+  debug: "temporary credentials for the assumed role are {{ ansible_temp_credentials }}"
+  
+- name: use temporary credentials for tagging an instance in account 123456789012 
+  ec2_tag: resource=i-xyzxyz01 region=us-west-1 state=present
+    args:
+      aws_access_key: "{{ ansible_temp_credentials.access_key }}"
+      aws_secret_key: "{{ ansible_temp_credentials.secret_key }}"
+      security_token: "{{ ansible_temp_credentials.session_token }}"
+    
+      tags:
+        Test: value
+'''
+
+import sys
+import time
+
+try:
+    import boto.sts
+    
+except ImportError:
+    print "failed=True msg='boto required for this module'"
+    sys.exit(1)
+
+def sts_connect(module):
+
+    """ Return an STS connection"""
+
+    region, ec2_url, boto_params = get_aws_connection_info(module)
+
+    # If we have a region specified, connect to its endpoint.
+    if region:
+        try:
+            sts = connect_to_aws(boto.sts, region, **boto_params)
+        except boto.exception.NoAuthHandlerFound, e:
+            module.fail_json(msg=str(e))
+    # Otherwise, no region so we fallback to connect_sts method
+    else:
+        try:
+            sts = boto.connect_sts(**boto_params)
+        except boto.exception.NoAuthHandlerFound, e:
+            module.fail_json(msg=str(e))
+
+
+    return sts
+
+def assumeRole():
+    data = sts.assume_role()
+    return data
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            role_arn = dict(required=True),
+            role_session_name = dict(required=True),
+            duraction_seconds = dict(),

```

Comment:
Spelling mistake.  Should be duration.  This is repeated elsewhere so might want to do a find and replace